app/menus/DocumentMenu.tsx

```diff
@@ -95,8 +97,35 @@ function DocumentMenu({
   });
   const { t } = useTranslation();
   const isMobile = useMobile();
+  const [subscriptionsLoaded, onLoad, onError] = useBoolean(false);
```

Comment:
Using local state purposely here since `subscriptions.isLoaded` is set _once and for all_ documents.

Reply:
We have too many one-offs like this, it's time to create something reusable… how about a hook:

```typescript
import * as React from "react";
import { client } from "~/utils/ApiClient";

export default function useApiRequest<T>(
  path: string,
  params?: Record<string, any> | undefined
): {
  data: T | undefined;
  error: any;
  loading: boolean;
  request: () => Promise<void>;
} {
  const [data, setData] = React.useState();
  const [loading, setLoading] = React.useState(false);
  const [error, setError] = React.useState();

  const request = React.useCallback(async () => {
    setLoading(true);
    try {
      const response = await client.post(path, params);
      setData(response);
    } catch (err) {
      setError(err);
    } finally {
      setLoading(false);
    }
  }, [path, params]);

  return { data, loading, error, request };
}
```

Reply:
Designed as a generic hook, for this case actually it would be better if it took a promise instead of a `path` and `params` so that we can use the existing store methods like:

```typescript
const { data, loading, request } = useRequest(() => subscriptions.fetchPage({
  documentId: document.id,
  event: "documents.update",
}));

if (!data && !loading) {
  request();
}
```

Reply:
Yeah, tested and pushed. Seems to work fine.